rust/src/lib.rs

```diff
@@ -15,6 +15,8 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#![feature(specialization)]
```

Comment:
Can you explain how we are now using the specialization feature?

Reply:
This is used in `BufferBuilder` where we defined a trait `BufferBuilderTrait` (I'm not too happy with this name..) and implemented this separately for `BufferBuilder<T: ArrowNumericType>` and `BufferBuilder<BooleanType>`. Without this, we'd have to implement `PrimitiveArrayBuilder` separately for `ArrowNumericType` and `BooleanType`, with the same code.

Reply:
😢